starter-code/src/clue.js

```diff
@@ -1,94 +1,212 @@
+    occupation: "Socialité" 
+ };
+ 
+ var mrMustard = {
+    firstName: "Jack",
+    lastName: "Mustard",
+    color: "yellow",
+    description: "He is a former football player who tries to get by on his former glory",
+    age: "62",
+    image: "https://metrouk2.files.wordpress.com/2016/07/colonel-mustard.jpg",
+    occupation: "Retired Football player" 
+ };
+
+// weapons variables
+
+var rope = {
+    name: "rope", weight: "10"
+};
+
+var knife = {
+    name: "knife", weight: "8"
+};
+
+var candlestick = {
+    name: "candlestick", weight: "2"
+};
+
+var dumbbell = {
+    name: "dumbbell", weight: "30"
+};
+
+var poison = {
+    name: "poison", weight: "2"
+};
+
+var axe = {
+    name: "axe", weight: "15"
+};
+
+var bat = {
+    name: "bat", weight: "13"
+};
+
+var trophy = {
+    name: "trophy", weight: "25"
+};
+
+var pistol = {
+    name: "pistol", weight: "20"
+};
+
+
+ // rooms variables
+ 
+ var dinningRoom = {
+    name: "Dinning Room"
+};
+
+var conservatory = {
+    name: "Conservatory"
+};
+
+var kitchen = {
+    name: "Kitchen"
+};
+
+var study = {
+    name: "Study"
+};
+
+var library = {
+    name: "Library"
+};
+
+var billiardRoom = {
+    name: "Billiard Room"
+};
+
+var lounge = {
+    name: "Lounge"
+};
+
+var ballroom = {
+    name: "Ballroom"
+};
+
+var hall = {
+    name: "Hall"
+};
+
+var spa = {
+    name: "Spa"
+};
+
+var livingRoom = {
+    name: "Living Room"
+};
+
+var observatory = {
+    name: "Observatory"
+};
+
+var theater = {
+    name: "Theater"
+};
+
+var guestHouse = {
+    name: "Guest House"
+};
+
+var patio = {
+    name: "Patio"
+};
+
 
 // Characters Collection
 var charactersArray = [];
 
+charactersArray.push(mrGreen, drOrchid, profPlum, missScarlet, mrsPeacock, mrMustard);
+
 // Rooms' Collection
 var roomsArray = [];
 
+roomsArray.push(dinningRoom, conservatory, kitchen, study, library, billiardRoom, lounge, ballroom, hall, spa, livingRoom, observatory, theater, guestHouse, patio);
+
 // Weapons Collection
 var weaponsArray = [];
+
+weaponsArray.push(rope, knife, candlestick, dumbbell, poison, axe, bat, trophy, pistol);
+
+
+// Create random
+
+function randomSelector (myArray) {
+    var randomNumber = Math.floor(Math.random() * myArray.length);
+    return myArray[randomNumber];
+}
+
+randomSelector(roomsArray);
+
+
+// Create the mystery
+
+function pickMistery () {
+
+    var misteryCard = [];
+
+    var misteryChar = randomSelector(charactersArray);
+    var misteryWeapons = randomSelector(weaponsArray);
+    var misteryRooms = randomSelector(roomsArray);
+
+    misteryCard.push(misteryChar, misteryWeapons, misteryRooms);
+
+    return misteryCard;
+}
+
+pickMistery();
+
+var misteryEnvelope = pickMistery ();
+
+function revealMistery(misteryEnvelope) {
+    return (misteryEnvelope[0].firstName + " " + misteryEnvelope[0].lastName + " killed Mr.Boddy unsing the " + misteryEnvelope[1].name + " in the " + misteryEnvelope[2].name + "!!!!");
```

Comment:
The error here is because you changed the property names in characters objects, in the starter code, it looked like that : 
```
first_name:   Jacob
last_name:    Green
color:        green
description:  He has a lot of connections
age:          45
image:        https://pbs.twimg.com/profile_images/506787499331428352/65jTv2uC.jpeg
occupation:   Entrepreneur
```
Jasmine doesn't know what 'misteryEnvelope[0].firstName' and 'misteryEnvelope[0].lastName' are, when testing the function, it passed as an argument an array that looked like that : 
```
[
 { 
  first_name: 'Victor', 
  last_name: 'Plum' 
 }, 
{ name: 'poison' }, 
{ name: 'Billiard Room' }
]
```